app/models/document.rb

```diff
@@ -208,6 +214,12 @@ def save
   def publish!
     handle_remote_error do
       update_type = self.update_type || 'major'
+
+      if not_published?
+        self.change_note = "First published."
+        self.save
+      end
```

Comment:
Some document formats (dfid research output) allow editing first_published_at directly so checking for that doesn't seem safe.  Checking the current state for any value doesn't seem safe either because the transitions aren't linear (e.g. we can go from draft to unpublished without going through published).  The "safest" option seems to be checking for "First published" in the change note (the edgecase here would be some editor adding that as their change note before it actually was published, but that'd be pretty pathological).

Hopefully we can do https://trello.com/c/kyRoUV0h/158-add-a-state-history-field-to-presented-items-in-publishing-api-medium soon and can re-write `ever_been_published?` to interrogate the full state history.  I'll add a note to https://trello.com/c/LU6PckOO/182-fix-document-history-for-redrafts-of-unpublished-documents to make it clear that this method would need to be changed.

How does that sound?